doc/OutOfTree.md

```diff
@@ -0,0 +1,142 @@
+Out of Tree Tock
+================
+
+This guide covers best practices for maintaining subsytems not in the
+Tock master repository.
+
+_It is a work in progress. Comments and pull requests are appreciated!_
+
+
+Overview
+--------
+
+Tock aims to maintain a stable syscall ABI, but does not guarantee
+stability of kernel interfaces. There are two primary channels to stay
+abreast of Tock development:
+
+  - [tock-dev mailing list](https://groups.google.com/forum/#!forum/tock-dev):
+    Any major Tock changes will be announced via this list. The list
+    also support general Tock development, however it is relatively low
+    traffic (<1 email/day on average).
+  - [Tock GitHub](https://github.com/helena-project/tock/): All Tock
+    changes go through Pull Requests. Non-trivial changes will generally
+    wait at least one week to merge to allow for feedback.
+
+Finally, please don't hesitate to
+[ask for help](https://kiwiirc.com/client/irc.freenode.net/tock).
+
+
+Structure
+---------
+
+Usually it is easiest to keep a
+[submodule](https://git-scm.com/docs/git-submodule) of Tock in your
```

Comment:
We don't need to block on merging this (very useful and important and timely doc), but do we actually think this is easiest/best? Rather than just using a Cargo git dependency?